test/contexts_spec.lua

```diff
@@ -77,11 +77,7 @@ setup(function()
     --- @type string
     local treesitter_lang = exec_lua(
       [[
-    local ok, parser = pcall(vim.treesitter.get_parser, ...)
-    if not ok then
-      return nil
-    end
-    return parser:lang()
+    return require('nvim-treesitter.parsers').get_buf_lang(...)
```

Comment:
`nvim-treesitter` is going to be replaced with the `main` branch soon enough, so I'd like to minimise relying on any it's API.

Reply:
so looks like in main we register the filetype to lang mappings [immediately](https://sourcegraph.com/github.com/nvim-treesitter/nvim-treesitter@873e118e1228d43540e076024710bc873fe9b74f/-/blob/plugin/filetypes.lua?L64) instead of [lazily](https://github.com/nvim-treesitter/nvim-treesitter/blob/7bbed4b548d9a81c14351bda93d16336edefd067/lua/nvim-treesitter/parsers.lua#L59-L68) in which case we can use `vim.treesitter.language.get_lang` directly. we could change it to that anticipatorily, just that this PR won't work until main is swapped in

Reply:
I think we can safely switch to `main` in this repo even though it isn't default yet.

Reply:
looks like the `main` branch is using an incompatible `vim.validate` signature than the one the runner is using. the runner expects a table, but we're giving a string

Reply:
We can update the runner to 0.11.

Reply:
We only have 1 runner version and several test versions. The runner version should ideally be the latest release.